openerp/addons/base/tests/test_orm.py

```diff
@@ -166,6 +166,22 @@ def read_group(interval, domain=None):
                         ['date'], ['date:month', 'date:day'], lazy=False)
         self.assertEqual(len(rg), len(all_partners))
 
+    def test_m2m_store_trigger(self):
+        group_user = self.env.ref('base.group_user')
+
+        user = self.env['res.users'].create({
+            'name': 'test',
+            'login': 'test_m2m_store_trigger',
+            'groups_id': [(6, 0, [])],
+        })
+        self.assertTrue(user.share)
+
+        group_user.write({'users': [(4, user.id)]})
+        self.assertFalse(user.share)
```

Comment:
The magic happen in `_write` method which call store triggers twice: [before](https://github.com/odoo/odoo/blob/326325a23ffc189991623ac47c54b3b723c1af44/openerp/models.py#L3849) the write in db and [after](https://github.com/odoo/odoo/blob/326325a23ffc189991623ac47c54b3b723c1af44/openerp/models.py#L4038)
It then update all the records returned by these calls